### PR TITLE
LP-2953 Account Deletion & Delivered Webinar Updates

### PR DESCRIPTION
**Pull request by javeriaejaz-arbisoft**
[LP-2953](https://philanthropyu.atlassian.net/browse/LP-2953)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/635)

***

#### Description:
Add a description of your changes here.

***

## Things to Cross-check:

- [ ] Technical Docs updated if needed?
- [ ] [Domain Model](https://drive.google.com/file/d/1iRSqxDQQMXl6JzLw9nZtq1yGshwMBLTn/view?usp=sharing) updated if needed?
- [ ] [Core Changes Wiki](https://github.com/OmnipreneurshipAcademy/edx-platform/wiki/Core-Changes) updated if needed?

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3409/workflows/workflows_id/jobs/3409/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/helpers.py (100%)
openedx/adg/lms/registration_extension/tests/test_helpers.py (100%)
openedx/adg/lms/student/analytics.py (93.3%): Missing lines 44
openedx/adg/lms/student/tests/test_analytics.py (100%)
-------------
Total:   58 lines
Missing: 1 line
Coverage: 98%
-------------
```

**Comment by javeriaejaz-arbisoft** [comment]
Please Ignore the quality flow comments as those refer to the core files.

**Review by faizan-philu — APPROVED** [review]

File: openedx/adg/lms/student/analytics.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    Remove user previous data & suppress new data from segment API
+
+    Arguments:
+        user (User): User to remove data
+
+    Return:
+        boolean: True if the request is successful
+    """
+    if hasattr(settings, 'SEGMENT_BEARER_TOKEN') and settings.SEGMENT_BEARER_TOKEN:
```
Comment (aightmunam) [inline openedx/adg/lms/student/analytics.py:24]: Let's also add a check for `SEGMENT_WORKSPACE` here

File: openedx/core/djangoapps/user_api/accounts/api.py [hunk 2]
```diff
@@ -446,7 +447,7 @@ def get_email_existence_validation_error(email):
     :return: Validation error message.
 
     """
-    return _validate(_validate_email_doesnt_exist, errors.AccountEmailAlreadyExists, email)
+    return _validate(validate_email_exists_or_retired, errors.AccountEmailAlreadyExists, email)
```
Comment (aightmunam) [inline openedx/core/djangoapps/user_api/accounts/api.py:450]: let's rename this function as we are validating that the email should NOT already exist and should NOT be retired already. So maybe something like,`validate_email_not_retired_nor_exists`?

File: openedx/adg/lms/registration_extension/helpers.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+"""
+Helpers for registration_extension app
+"""
+from django.contrib.auth.models import User
```
Comment (aightmunam) [inline openedx/adg/lms/registration_extension/helpers.py:4]: Let's use `django.contrib.auth.get_user_model()` to get the User model

File: openedx/adg/lms/registration_extension/tests/test_helpers.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.adg.lms.registration_extension.constants import RETIRED_EMAIL_MESSAGE
+from openedx.adg.lms.registration_extension.helpers import validate_email_exists_or_retired
+from openedx.core.djangoapps.user_api import accounts
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_retired', [True, False],
+)
```
Comment (aightmunam) [inline openedx/adg/lms/registration_extension/tests/test_helpers.py:15]: Let's add a test for already existing email as well

**Review by aightmunam — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/registration_extension/tests/test_helpers.py (hunk 4 above)
Comment (javeriaejaz-arbisoft) [reply openedx/adg/lms/registration_extension/tests/test_helpers.py:15]: Already testing it.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3410/workflows/workflows_id/jobs/3410/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/helpers.py (100%)
openedx/adg/lms/registration_extension/tests/test_helpers.py (100%)
openedx/adg/lms/student/analytics.py (87.5%): Missing lines 44-45
openedx/adg/lms/student/tests/test_analytics.py (100%)
-------------
Total:   60 lines
Missing: 2 lines
Coverage: 96%
-------------
```

File: openedx/adg/lms/student/analytics.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {settings.SEGMENT_BEARER_TOKEN}'
+        }
+
+        response = requests.request('POST', url, headers=headers, data=data)
+        log.info(f'Deleting User {user.username} data from Segment API, response code: {response.status_code}')
+        return response.status_code == HTTP_200_OK
+
+    log.info(f'Segment Bearer Token or Workspace not found!')
```
Comment (aightmunam) [inline openedx/adg/lms/student/analytics.py:44]: Perhaps a little more info.  
Cannot remove segment data for user `email` failed. Segment Bearer Token or Workspace not found!

**Review by aightmunam — APPROVED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3411/workflows/workflows_id/jobs/3411/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/helpers.py (100%)
openedx/adg/lms/registration_extension/tests/test_helpers.py (100%)
openedx/adg/lms/student/analytics.py (87.5%): Missing lines 44-45
openedx/adg/lms/student/tests/test_analytics.py (100%)
-------------
Total:   60 lines
Missing: 2 lines
Coverage: 96%
-------------
```

**Review by aightmunam — APPROVED** [review]

**Review by faizan-philu — APPROVED** [review]

**Pull request merged**